lib/words_report.rb

```diff
@@ -0,0 +1,28 @@
+class WordsReport
+  attr_reader :words
+
+  # Arbitrary definition of popular
+  POPULAR_COUNT = 5
+
+  def initialize(counts)
+    @counts = counts
+    @words = words_by_frequency
+  end
+
+  def popular_words(popular_count = POPULAR_COUNT)
+    @words.select { |word| @counts[word] >= popular_count }
+  end
+
+  def report(popular_count = POPULAR_COUNT)
+    popular_words(popular_count).each { |word| printf "%-12s: %3d\n", word, @counts[word] }
+  end
+
+  private
+
+  def words_by_frequency
+    @words = @counts.keys.sort do |a,b|
+      first = @counts[b] <=> @counts[a]
+      first == 0 ? a <=> b : first
```

Comment:
It would be a lot easier to read if we didn't need the words with the same count sorted alphabetically. Do we really need that?